web/src/main/java/org/fao/geonet/services/metadata/MetadataCacheSeeder.java

```diff
@@ -0,0 +1,397 @@
+            }
+            if (Log.isDebugEnabled(Geonet.CACHESEEDER))
+                Log.debug(Geonet.CACHESEEDER, "metadata cache seeder job ended");
+        }
+    }
+
+    
+    /**
+     * This method is responsible of generating a sitemap.xml file in the
+     * cache path folder.
+     * 
+     * @param Iterator<File> cachedFiles the list of files to include
+     * to the sitemap.xml.
+     */
+    private void generateSitemapXml(Iterator<File> cachedFiles) {
+        Element urlset = new Element("urlset", SITEMAP_NS);
+        while (cachedFiles.hasNext()) {
+            File f = cachedFiles.next();
+            String lastMod = new ISODate(f.lastModified()).toString();
+            
+            String strf = f.getAbsolutePath();
+            Element url = new Element("url",SITEMAP_NS);
+            url.addContent(new Element("loc", SITEMAP_NS).setText(urlPath + "/" + FilenameUtils.getName(strf)));
+            url.addContent(new Element("changefreq", SITEMAP_NS).setText(changeFreq));
+            url.addContent(new Element("lastmod", SITEMAP_NS).setText(lastMod));
+            urlset.addContent(url);
+        }
+        
+        try {
+            FileUtils.writeStringToFile(new File(cachePath, "sitemap.xml"), Xml.getString(urlset));
+        } catch (IOException e) {
+            Log.error(Geonet.CACHESEEDER, "Unable to generate the sitemap.xml file: ", e);
+        }
+    }
+
+    /**
+     * This method generates a single html file from a given UUID.
+     * 
+     * @param dbms a DBMS resource to access datas from the database.
+     * @param curUuid the UUID of the metadata.
+     *
+     * @throws Exception
+     */
+    private void generateHtmlFile(Dbms dbms, String curUuid) throws Exception {
+        // This does globally the same as Format.java
+        // In a more concise way.
+        
+        Element root = new Element("root");
+        Element md = Xml.loadString(getMetaData(dbms, curUuid), false);
+
+        root = root.addContent(md);
+
+        Element transformed = Xml.transform(root, xslPath);
+        FileUtils.writeStringToFile(new File(this.cachePath, curUuid + ".html"), Xml.getString(transformed));
+    }
+
+    private String getMetaData(Dbms dbms, String curUuid) throws Exception {
+        Element resp = dbms.select("SELECT data FROM metadata WHERE uuid = ? LIMIT 1; ", curUuid);
+        return resp.getChild("record").getChildText("data");
+    }
+
+    private Date getMetadataChangeDate(Dbms dbms, String curUuid) {
+        Date d = new Date();
+        try {
+            Element resp = dbms.select(
+                    "SELECT changedate FROM metadata WHERE uuid = ? LIMIT 1; ",
+                    curUuid);
+            String strDate = resp.getChild("record").getChildText("changedate");
+            String[] patterns = { DateFormatUtils.ISO_DATETIME_FORMAT
+                    .getPattern() };
+            strDate = strDate.substring(0, 19);
+            d = DateUtils.parseDate(strDate, patterns);
```

Comment:
Not sure if it applies for this version of GN but in develop there is a class: IsoDate that handles parsing the dates in metadata and database